lib/blocks.php

```diff
@@ -45,3 +45,71 @@ function gutenberg_reregister_core_block_types() {
 	}
 }
 add_action( 'init', 'gutenberg_reregister_core_block_types' );
+
+/**
+ * Registers a new block style.
+ *
+ * @param string $block_name       Block type name including namespace.
+ * @param array  $style_properties Array containing the properties of the style name, label, style (name of the stylesheet to be enqueued), inline_style (string containing the CSS to be added).
+ *
+ * @return boolean True if the block style was registered with success and false otherwise.
+ */
+function register_block_style( $block_name, $style_properties ) {
+	return WP_Block_Styles_Registry::get_instance()->register( $block_name, $style_properties );
+}
+
+/**
+ * Unregisters a block style.
+ *
+ * @param string $block_name       Block type name including namespace.
+ * @param array  $block_style_name Block style name.
+ *
+ * @return boolean True if the block style was unregistered with success and false otherwise.
+ */
+function unregister_block_style( $block_name, $block_style_name ) {
+	return WP_Block_Styles_Registry::get_instance()->unregister( $block_name, $block_style_name );
+}
+
+/**
+ * Function responsible for enqueuing the assets required for block styles functionality.
+ */
+function enqueue_block_styles_assets() {
+	$block_styles = WP_Block_Styles_Registry::get_instance()->get_all_registered();
+
+	foreach ( $block_styles as $block_name => $styles ) {
+		foreach ( $styles as $style_properties ) {
+			if ( isset( $style_properties['style_handle'] ) ) {
+				wp_enqueue_style( $style_properties['style_handle'] );
+			}
+			if ( isset( $style_properties['inline_style'] ) ) {
+				wp_add_inline_style( 'wp-block-library', $style_properties['inline_style'] );
+			}
+
+			$client_style_properties = $style_properties;
+			if ( isset( $client_style_properties['style_handle'] ) ) {
+				unset( $client_style_properties['style_handle'] );
+			}
+			if ( isset( $client_style_properties['inline_style'] ) ) {
+				unset( $client_style_properties['inline_style'] );
+			}
+
+			wp_add_inline_script(
+				'wp-blocks',
+				sprintf(
+					implode(
+						"\n",
+						array(
+							'( function() {',
+							'	wp.blocks.registerBlockStyle( \'%s\', %s );',
+							'} )();',
+						)
+					),
+					$block_name,
+					wp_json_encode( $client_style_properties )
```

Comment:
Could we pass the right attributes directly `array( 'key' => $values )` instead of having to unset properties as it could break if we add new properties in the future.